torch/_inductor/codecache.py

```diff
@@ -379,6 +388,163 @@ def write_atomic(path: str, content: Union[str, bytes]) -> None:
+        """
+        Pickle an object using the FxGraphCachePickler.
+        """
+        with io.BytesIO() as stream:
+            pickler = FxGraphCachePickler(stream)
+            pickler.dump(obj)
+            return stream.getvalue()
+
+
+@dataclasses.dataclass
+class OrderedSetHolder:
+    items: List[Any]
+
+
+class FxGraphHashDetails:
+    """
+    Object to capture all the details for a compiled FX graph relevant to computing
+    a safe and stable cache key.
+    """
+
+    # Excluded kwargs param that are not stable between runs
+    EXCLUDED_KWARGS = ["graph_id"]
+
+    def __init__(self, fx_args: List[Any], fx_kwargs: Dict[str, Any]) -> None:
+        self.fx_args = fx_args
+
+        # Order kwargs so hashing is stable to changes in kwarg order.
+        self.fx_kwargs = {}
+        for k in sorted(fx_kwargs):
+            if k not in self.EXCLUDED_KWARGS:
+                if type(fx_kwargs[k]) is set:
+                    # Special case to handle set params. Python sets can't be
+                    # ordered, so sort the elements and store them in a proxy.
+                    self.fx_kwargs[k] = OrderedSetHolder(sorted(fx_kwargs[k]))
+                else:
+                    self.fx_kwargs[k] = fx_kwargs[k]
+
+        self.torch_version = torch.__version__
+        self.triton_version = triton.runtime.version_key()
+        self.inductor_config = config.save_config()  # type: ignore[attr-defined]
+
+
+def compiled_fx_graph_hash(fx_args: List[Any], fx_kwargs: Dict[str, Any]) -> str:
+    """
+    Generate a unique hash of the FX graph for caching.
+    """
+    details = FxGraphHashDetails(fx_args, fx_kwargs)
+    serialized_data = FxGraphCachePickler.dumps(details)
+    return (
+        "f"
+        + base64.b32encode(hashlib.sha256(serialized_data).digest())[:51]
+        .decode("utf-8")
+        .lower()
+    )
+
+
+class FxGraphCache:
+    """
+    Supports caching and reusing compiled Fx graphs.
+    """
+
+    # TODO(slarsen): Investigate whether it's beneficial to store compiled graphs
+    # in an in-memory cache after loading from disk.
+    @classmethod
+    def save_graph(cls, key: str, compiled_graph: CompiledFxGraph):
+        disk_compiled_graph = copy(compiled_graph)
+        # Important as compiled models are not pickleable
+        # TODO: Check status of PR #101651 as that might change the above statement
+        disk_compiled_graph.compiled_artifact = None
+        write(pickle.dumps(disk_compiled_graph), "cg", extra=key, hash_type="cg")
+
+    @classmethod
+    def load_graph(cls, cg_path: str) -> CompiledFxGraph:
+        with open(cg_path, "rb") as f:
+            return pickle.load(f)
+
+    @classmethod
+    def load(
+        cls,
+        compile_fx_fn: Callable[..., Any],
+        fx_args: List[Any],
+        fx_kwargs: Dict[str, Any],
+    ):
+        from filelock import FileLock
+
+        try:
+            key = compiled_fx_graph_hash(fx_args, fx_kwargs)
+        except Exception as ex:
```

Comment:
@eellison Was just trying to be defensive in case the hashing fails. I have not seen it fail in my recent testing (running as many benchmarks as I can), however. I can certainly remove if you think it's better to crash.

Reply:
Let's remove this then, especially since this isn't on by default yet anyway.